duke/src/main/java/DeleteCommand.java

```diff
@@ -0,0 +1,22 @@
+import java.util.ArrayList;
+
+class DeleteCommand extends Command {
+	public DeleteCommand(String command, String remainingCommand) {
+		super(command, remainingCommand);
+	}
+
+	public void execute(TaskList tasks, Ui ui) throws DukeException {
+		ArrayList<Task> list = tasks.getList();
+		if (list.isEmpty()) {
+			throw new DukeException("You have no task to delete!");
+		} else if (remainingCommand.isEmpty()) {
+			throw new DukeException("Which task do you want to delete?");
+		}
+		
+		int index = Integer.valueOf(remainingCommand); // assuming only has number left
+		Task task = list.get(index - 1);
+		list.remove(task);
```

Comment:
I feel like the deletion should be done by TaskList itself! In case you need to change its implementation, you would not have to refactor this class too